docs/dev_cycle.part-01.rst

```diff
@@ -0,0 +1,242 @@
+To learn about the greeter example see the :ref:`Quickstart <greete_quickstart>`.
+
+We will need a local private blockchain. This local blockchain
+is an excellent tool for development: it runs on your machine, does not take the time to sync the real blockchain, does not
+costs real gas and fast to respond.
+Yet the local chain works with the same Ethereum protocol. If a contract runs
+on the local chain, it should run on mainnet as well. See :ref:`runing_local_blockchain` .
+
+.. note::
+    If you are familiar to web development, then running a local blockchain is
+    similar to running a local website on 127.0.0.1, before publishing it to the internet.
+
+We will create a a local chain we'll name "horton":
+
+.. code-block:: bash
+
+    $ populus chain new horton
+    $ chains/horton/./init_chain.sh
+
+    INFO [10-14|12:31:31] Allocated cache and file handles         database=/home/mary/projects/donations/chains/horton/chain_data/geth/chaindata cache=16 handles=16
+    INFO [10-14|12:31:31] Writing custom genesis block
+    INFO [10-14|12:31:31] Successfully wrote genesis state         database=chaindata                                                                        hash=faa498…370bf1
+    INFO [10-14|12:31:31] Allocated cache and file handles         database=/home/mary/projects/donations/chains/horton/chain_data/geth/lightchaindata cache=16 handles=16
+    INFO [10-14|12:31:31] Writing custom genesis block
+    INFO [10-14|12:31:31] Successfully wrote genesis state         database=lightchaindata                                                                        hash=faa498…370bf1
+
+Everything is ready.
+
+Add a Contract
+----------------
+
+Ok, time to add a new contract.
+
+.. code-block:: shell
+
+    $ nano contracts/Donator.sol
+
+
+.. note::
+
+    You can work with your favourite IDE. Check for Solidity extention/package. Atom.io
+    has some nice Solidity packages.
+
+
+In this example we will work with a very simple contract that accepts donations for later use.
+The contract will also handle the donations value in USD.
+
+Since the ETH/USD exchange rate fluctates, typically upward, we want to track not only how much ETH the contract collected,
+but also the accumulating USD value of the donations *at the time of the donation*.
+If the ETH rate is rising, then we will probably see smaller donations
+in terms of Ether, but similar donations in terms of USD.
+
+In other words, two donations of say $30 will have different amounts in ETH if the exchange rate changed between
+the donations. As a simple solution, we will ask donators to provide the effective ETH/USD exchange rate when they send their (hopefully generous) donations.
+
+Here is the new contract code:
+
+.. code-block:: solidity
+
+    pragma solidity ^0.4.0;
+
+    /// TUTORIAL CONTRACT DO NOT USE IN PRODUCTIO
+    /// @title Donations collecting contract
```

Comment:
Sphinx has a directive called a *literal include* which lets you include the contents of a file inline.  This would potentially be a good tool for these inlined source files.  We could include them in the `ASSETS_DIR`, have some tests that verified that they compiled, and then just inline them here.  That would help make sure these tutorials stayed functional.